File: src/Application Specific/Classes/Entity Collections/ContractsOriginal/ContractOriginalSiteAsset/ContractOriginalSiteAssetValidator.vb
Namespace Entity

    Namespace ContractOriginalSiteAssets

        Public Class ContractOriginalSiteAssetValidator
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oContractSiteAsset As ContractOriginalSiteAsset)

                'make sure that contact object is valid
                If oContractSiteAsset.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oContractSiteAsset.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If



                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/ContractsOriginal/ContractOriginalSiteAsset/ContractOriginalSiteAssetValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace ContractOriginalSiteAssets
    {
        // make sure that contact object is valid
        public class ContractOriginalSiteAssetValidator : BaseValidator
        {
            public void Validate(ContractOriginalSiteAsset oContractSiteAsset)
            {

                // make sure that contact object is valid
                if (oContractSiteAsset.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oContractSiteAsset.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}